File: valid-parentheses/validParenthesis.js
function isValid(str) {
    let stack = []
    let opens = { '(': ')', '{': '}', '[': ']' }

    for (const brace of str) {
        if (brace in opens) {
            stack.push(opens[brace])
        } else if ((brace === ')' || brace === ']' || brace == '}')
            && (stack.length === 0 || stack.pop() !== brace)) {
            return false
        }
    }

    return stack.length === 0
}

File: valid-parentheses/ValidParenthesis.java
import java.util.Stack;

public class ValidParenthesis {
    public static boolean isValid(String str) {
        Stack<Character> stack = new Stack<>();

        for (char brace : str.toCharArray()) {
            switch (brace) {
                case '(':
                    stack.push(')');
                    break;
                case '{':
                    stack.push('}');
                    break;
                case '[':
                    stack.push(']');
                    break;
                case ')':
                case '}':
                case ']':
                    if (stack.isEmpty() || stack.pop() != brace) {
                        return false;
                    }
                    break;
                default:
                    break;
            }
        }

        return stack.isEmpty();
    }
}
